src/core_init_atmosphere/mpas_stack.F

```diff
@@ -0,0 +1,273 @@
+module mpas_stack
+    
+   implicit none
+
+   private
+
+   ! Public Subroutines and Structures
+   public :: mpas_stack_is_empty
+   public :: mpas_stack_push
+   public :: mpas_stack_pop
+   public :: mpas_stack_free
+
+   public :: node, payload_t
+
+   type payload_t
+   end type payload_t
+
+   type node
+      type (node), pointer :: next => null()
+      class (payload_t), pointer :: payload => null()
+   end type node
+
+   !***********************************************************************
+   !
+   !  module mpas_stack
+   !
+   !> \brief   MPAS KD-Tree module
```

Comment:
Should this be `stack` rather than `KD-Tree`?

Reply:
Yes!